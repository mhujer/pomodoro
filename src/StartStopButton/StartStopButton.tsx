import React from 'react';

type Props = {
    handleButtonClick: () => void;
}

const StartStopButton: React.FC<Props> = ({ handleButtonClick }) => {

    return (
        <>
            <button onClick={handleButtonClick}>Start!</button>
        </>
    );

};

export default StartStopButton;
