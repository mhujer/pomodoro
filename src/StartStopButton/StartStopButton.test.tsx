import StartStopButton from './StartStopButton';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('StartStopButton', () => {
    it('renderes isPomodoroActive=false', () => {
        const domTree = renderer.create(<StartStopButton isPomodoroActive={false}
                                                         handleStart={jest.fn()}
                                                         handleStop={jest.fn()} />).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('renderes isPomodoroActive=true', () => {
        const domTree = renderer.create(<StartStopButton isPomodoroActive={true}
                                                         handleStart={jest.fn()}
                                                         handleStop={jest.fn()} />).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('renders and clicks Start', () => {
        const startClickSpy = jest.fn();
        const stopClickSpy = jest.fn();

        render(<StartStopButton isPomodoroActive={false}
                                handleStart={startClickSpy}
                                handleStop={stopClickSpy} />);

        fireEvent.click(screen.getByText('Start!'));

        expect(startClickSpy).toHaveBeenCalled();
        expect(stopClickSpy).not.toHaveBeenCalled();

    });

    it('renders and clicks Stop', () => {
        const startClickSpy = jest.fn();
        const stopClickSpy = jest.fn();

        render(<StartStopButton isPomodoroActive={true}
                                handleStart={startClickSpy}
                                handleStop={stopClickSpy} />);

        fireEvent.click(screen.getByText('Stop!'));

        expect(startClickSpy).not.toHaveBeenCalled();
        expect(stopClickSpy).toHaveBeenCalled();

    });
});
