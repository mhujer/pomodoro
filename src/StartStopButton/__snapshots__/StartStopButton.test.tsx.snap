// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderes isPomodoroActive=false 1`] = `
<button
  onClick={[MockFunction]}
>
  Start!
</button>
`;

exports[`renderes isPomodoroActive=true 1`] = `
<button
  onClick={[MockFunction]}
>
  Stop!
</button>
`;
